src/Shipping/ShippingSuggestionService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Shipping;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WC;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class ShippingSuggestionService
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Shipping
+ *
+ * @since x.x.x
+ */
+class ShippingSuggestionService implements Service {
+
+	/**
+	 * @var ShippingZone
+	 */
+	protected $shipping_zone;
+
+	/**
+	 * @var WC
+	 */
+	protected $wc;
+
+	/**
+	 * ShippingSuggestionService constructor.
+	 *
+	 * @param ShippingZone $shipping_zone
+	 * @param WC           $wc
+	 */
+	public function __construct( ShippingZone $shipping_zone, WC $wc ) {
+		$this->shipping_zone = $shipping_zone;
+		$this->wc            = $wc;
+	}
+
+	/**
+	 * Get shipping rate suggestions.
+	 *
+	 * @param string $country_code
+	 *
+	 * @return array A multidimensional array of shipping rate suggestions. {
+	 *     Array of shipping rate suggestion arguments.
+	 *
+	 *     @type string $country  The shipping country.
+	 *     @type string $method   The shipping method (e.g. 'flat_rate').
+	 *     @type string $currency The suggested rate currency (this is the same as the store's currency).
+	 *     @type float  $rate     The cost of the shipping method.
+	 *     @type array  $options  Array of options for the shipping method.
+	 * }
+	 */
+	public function get_suggestions( string $country_code ): array {
+		$location_rates = $this->shipping_zone->get_shipping_rates_grouped_by_country( $country_code );
+
+		$suggestions    = [];
+		$free_threshold = null;
+		foreach ( $location_rates as $location_rate ) {
+			$serialized = $location_rate->jsonSerialize();
+
+			// Check if there is a conditional free shipping rate (with minimum order amount).
+			// We will set the minimum order amount as the free shipping threshold for other rates.
+			$shipping_rate = $location_rate->get_shipping_rate();
+			if ( $shipping_rate->is_free() && $shipping_rate->has_min_order_amount() ) {
+				$free_threshold = $shipping_rate->get_min_order_amount();
+
+				// Ignore the conditional free rate if there are other rates.
+				if ( count( $location_rates ) > 1 ) {
```

Comment:
> I am not really sure why this is necessary, for example if we have a free shipping with a min. order amount and no other rates, the rate is skipped anyway.

This is for the inverse condition, meaning that we have conditional free shipping rates AND other rates. What this part does is that it gets the `min_order_amount` value from a conditional free shipping rate and adds it as a `free_shipping_threshold` to all the other rates. When we add that value to the other rates, we need to remove the free shipping rate from the list because it's already implied by that `free_shipping_threshold` value in other rates.

This is just a data format change so that it can be returned the way UI expects it.

Instead of returning this data:

```JSON
    {
        "country": "US",
        "method": "flat_rate",
        "currency": "USD",
        "rate": 20
    }
    {
        "country": "US",
        "method": "flat_rate",
        "currency": "USD",
        "rate": 0,
        "min_order_amount": 100
    }
```

We would return the following in the API response:

```JSON
    {
        "country": "US",
        "method": "flat_rate",
        "currency": "USD",
        "rate": 20,
        "options": {
            "free_shipping_threshold": 100
        }
    }
```

Reply:
> I think we could remove if ( count( $location_rates ) > 1 ) but I do not know if I am missing something?

I rather keep this condition because if we remove it then the code here would depend on the conditions defined in other classes with no way to check if the logic is as expected. This condition helps make sure the conditional free rate is always ignored when its parameters are applied to the other rates.

Reply:
Thanks @nima-karimi that makes sense 👍